src/multiselect.js

```diff
@@ -190,6 +191,24 @@ export class Multiselect {
     // on Block Selected
     if (e.type === Blockly.Events.SELECTED) {
       this.controls_.updateMultiselect();
+    // on Block field changed
+    } else if (e.type === Blockly.Events.CHANGE &&
+        e.element === 'field' && e.recordUndo &&
+        this.blockSelection_.has(e.blockId)) {
+      Blockly.Events.setGroup(true);
```

Comment:
We may want to check if a group is currently active. If so, we should continue with the current group and not call setGroup(false) at the end of this scope. It might also be worth wrapping the updates in a try-finally block so that even if an exception is thrown we reset the group state correctly.

Reply:
Thanks for the suggestion, fixed!